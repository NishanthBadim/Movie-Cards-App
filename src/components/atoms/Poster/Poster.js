import React from 'react'
import ModalVideo from 'react-modal-video'
import '../../../index.css';
import styled from 'styled-components'

class Poster extends React.Component {

  constructor (props) {
    super(props)
    this.state = {
      isOpen: false
    }
  }

  openModal = () => {
    this.setState({isOpen: true})
  }

  Button = styled.button`
  border:none;
  position: absolute;

  margin-left: 50px;
  margin: 0;
  margin-top: -110px;
  margin-left: 70px;
  background: transparent;
  color:rgb(212, 212, 212);
  transition: all 0.3s ease-in-out;

  .fa{
    margin-left: 50px;
    font-size: 50px;
  }

  &:hover,&:focus {
    cursor: pointer;
    background: transparent;
    font-size:60px;
    outline: none;
    color:white;
  }
  `
  render () {
    return (
      <div>
        <img src={this.props.image} alt="movie-poster" style={{height:'170px',width:'298px'}}/>
        <div className="modalVideo" data-testid="modal">
            <ModalVideo  channel='youtube' isOpen={this.state.isOpen} videoId={this.props.video}  onClose={() => this.setState({isOpen: false})} />
        </div>
        <this.Button onClick={this.openModal} className="button" id="videolink">
                <i className="fa fa-play-circle" arial-hidden="true"></i>
        </this.Button>
      </div>
    )
  }
}
export default Poster;