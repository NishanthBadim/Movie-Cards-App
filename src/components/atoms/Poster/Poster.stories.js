import React from 'react';
import Furious from '../../../images/Furious7.jpg';
import { withKnobs, text, files } from '@storybook/addon-knobs/react'
import Poster from './Poster'

export default {
    component: Poster,
    title: "Poster",
    decorators: [withKnobs]
}

export const Default = () => {
    return (
        <Poster
            image={files('Image', '.jpg', Furious)}
            video={text("VideoID", "Skpu5HaVkOc")}
        />
    )
}

