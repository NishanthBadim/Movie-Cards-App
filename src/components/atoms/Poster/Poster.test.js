import React from 'react';
import {create} from 'react-test-renderer'
import "@testing-library/jest-dom/extend-expect";
import { render, screen, getByRole, fireEvent } from "@testing-library/react";
import Poster from './Poster'

test("Match the snapshot - Poster",() => {
    const input=create(<Poster />);
    expect(input.toJSON()).toMatchSnapshot();
})

test("Image is present", () => {
    render(<Poster/>);
    const image=screen.getByAltText("movie-poster");
    expect(image).toBeInTheDocument();

})
test("Video is popping up", () => {
    const {getByRole, getByTestId} = render(<Poster/>)
    fireEvent.click(getByRole('button'))
    expect(getByTestId('modal')).toBeInTheDocument();
})
test("Button is present", () => {
    render(<Poster/>);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
  });
