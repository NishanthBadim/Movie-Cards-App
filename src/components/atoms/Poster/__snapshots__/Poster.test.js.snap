// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot - Poster 1`] = `
<div>
  <img
    alt="movie-poster"
    style={
      Object {
        "height": "170px",
        "width": "298px",
      }
    }
  />
  <div
    className="modalVideo"
    data-testid="modal"
  />
  <button
    className="sc-AxjAm gBHxQp button"
    id="videolink"
    onClick={[Function]}
  >
    <i
      arial-hidden="true"
      className="fa fa-play-circle"
    />
  </button>
</div>
`;
